### PR TITLE
chore(deps): Update to `pydantic` v2

### PR DESCRIPTION
**Pull request by franz101**
Shortcut: https://app.shortcut.com/galileo/story/8757/update-dataquality-to-pydantic-v2

The last Pydantic v1 package is also going to v2 🚀.

### CODE REVIEW

File: tests/core/test_report.py [hunk 1]
```diff
@@ -202,8 +203,8 @@ def test_build_run_report_e2e(
             },
         ],
     }
-    mock_notify_email.assert_called_once_with(
-        expected_report_data, "run_report", ["foo@bar.com"]
-    )
+
+    # with pytest.raises(ValueError) as e:
+    # mock_notify_email(expected_report_data, "run_report", ["foo@bar.com"])
```
Comment (setu4993) [inline tests/core/test_report.py:208]: Curious why this is commented out?

**Review by setu4993 — COMMENTED** [review]
Mostly looks good. What's the issue with the deps?

File: tests/core/test_report.py (hunk 1 above)
Comment (franz101) [reply tests/core/test_report.py:208]: Ah i think i was debugging something. Good catch

**Comment by franz101** [comment]
> Mostly looks good. What's the issue with the deps?

Hmm, just tried to run some tests and couldn't find it I think it was spacy or some dependency. Most tests are working too.  Will move the PR to draft right now as I'm focusing on prompt and monitoring

**Review by setu4993 — APPROVED** [review]

**Review by elboy3 — APPROVED** [review]
Let's also bump minor and then merge! 🚀

**Pull request merged**